### PR TITLE
[AF-1733] Turn off the ssl verification when cloning a git project

### PR DESCRIPTION
**Pull request by ederign**
A version of https://github.com/kiegroup/appformer/pull/555 with tests added.

@ghjboss could you please also take a look?

Docs: https://github.com/kiegroup/kie-docs/pull/1261

### CODE REVIEW

**Comment by kiereleaseuser2** [comment]
Build finished. 5032 tests run, 11 skipped, 0 failed.

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/commands/CreateRepository.java [hunk 1]
```diff
@@ -32,28 +34,35 @@
     private final File repoDir;
     private final File hookDir;
     private final KetchLeaderCache leaders;
+    private final boolean sslVerify;
 
-    public CreateRepository(final File repoDir) {
+    public CreateRepository(final File repoDir,
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/commands/CreateRepository.java]: i'd like to preserve the original constructor and default sslVerify to false

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/commands/Fork.java [hunk 2]
```diff
@@ -39,13 +39,15 @@
...
+    private final boolean sslVerify;
 
     public Fork(final File parentFolder,
                 final String source,
                 final String target,
                 final CredentialsProvider credentialsProvider,
                 final KetchLeaderCache leaders,
-                final File hookDir) {
+                final File hookDir,
+                final boolean sslVerify) {
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/commands/Fork.java]: same about constructor

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/Git.java [hunk 3]
```diff
@@ -49,62 +49,80 @@
 
 public interface Git {
 
-    static Git createRepository(final File repoDir) {
+    static Git createRepository(final File repoDir,
+                                final boolean sslVerify) {
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/Git.java]: can't we preseve the orignal and default sslVerify to false?

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/Git.java [hunk 4]
```diff
@@ -49,62 +49,80 @@
...
     }
 
     static Git clone(final File repoDest,
                      final String origin,
                      final boolean isMirror,
                      final CredentialsProvider credential,
                      final KetchLeaderCache leaders,
-                     final File hookDir) {
+                     final File hookDir,
+                     final boolean sslVerify) {
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/Git.java]: same comments about ovearload/constructor

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/commands/Clone.java [hunk 5]
```diff
@@ -53,7 +54,8 @@ public Clone(final File directory,
                  final boolean isMirror,
                  final CredentialsProvider credentialsProvider,
                  final KetchLeaderCache leaders,
-                 final File hookDir) {
+                 final File hookDir,
+                 final boolean sslVerify) {
```
Comment (porcelli) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/commands/Clone.java]: same about keep constructor and default sslVerify to false

File: uberfire-ssh/uberfire-ssh-backend/src/test/resources/.security/pkeys/john/.john-key.pub.meta [hunk 6]
```diff
@@ -1 +1 @@
-{"name":"John Key one","creationDate":"Jul 19, 2018 6:09:51 PM","lastTimeUsed":"Oct 11, 2018 11:19:19 AM"}
\ No newline at end of file
+{"name":"John Key one","creationDate":"Jul 19, 2018 6:09:51 PM","lastTimeUsed":"Nov 29, 2018 12:13:38 AM"}
```
Comment (porcelli) [inline uberfire-ssh/uberfire-ssh-backend/src/test/resources/.security/pkeys/john/.john-key.pub.meta]: why change this?

File: uberfire-ssh/uberfire-ssh-backend/src/test/resources/custom_security/pkeys/john/.john-key.pub.meta [hunk 7]
```diff
@@ -1 +1 @@
-{"name":"John Key one","creationDate":"Jul 19, 2018 6:09:51 PM","lastTimeUsed":"Nov 16, 2018 9:25:49 AM"}
\ No newline at end of file
+{"name":"John Key one","creationDate":"Jul 19, 2018 6:09:51 PM","lastTimeUsed":"Oct 11, 2018 11:19:19 AM"}
```
Comment (porcelli) [inline uberfire-ssh/uberfire-ssh-backend/src/test/resources/custom_security/pkeys/john/.john-key.pub.meta]: why change this?

File: uberfire-ssh/uberfire-ssh-backend/src/test/resources/custom_security/pkeys/katy/.katy-key-two.pub.meta [hunk 8]
```diff
@@ -1 +1 @@
-{"name":"Katy Key two","creationDate":"Jul 19, 2018 6:09:51 PM","lastTimeUsed":"Nov 16, 2018 9:25:49 AM"}
\ No newline at end of file
+{"name":"Katy Key two","creationDate":"Jul 19, 2018 6:09:51 PM","lastTimeUsed":"Nov 29, 2018 12:13:38 AM"}
```
Comment (porcelli) [inline uberfire-ssh/uberfire-ssh-backend/src/test/resources/custom_security/pkeys/katy/.katy-key-two.pub.meta]: why change this?

**Comment by ederign** [comment]
@porcelli updated as requested.

**Comment by kiereleaseuser2** [comment]
Build finished. 5038 tests run, 11 skipped, 0 failed.

**Comment by domhanak** [comment]
jenkins retest this

**Comment by ederign** [comment]
hi @domhanak do you have time to test this?

**Comment by domhanak** [comment]
jenkins execute full downstream build

**Comment by domhanak** [comment]
@ederign The FDB I run yesterday got stuck, please rerun FDB once you resolve the conflict from #564 
I will take a look in the morning.

**Comment by ederign** [comment]
@domhanak now without conflicts

Jenkins execute full downstream build

**Comment by domhanak** [comment]
Jenkins execute full downstream build

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/commands/CreateRepository.java [hunk 9]
```diff
@@ -79,6 +112,13 @@ public CreateRepository(final File repoDir,
 
             final org.eclipse.jgit.api.Git git = new org.eclipse.jgit.api.Git(repo);
 
+            if(!sslVerify) {
+                StoredConfig config = repo.getConfig();
+                config.setBoolean( "http", null, "sslVerify", false );
+                config.save();
+            }
```
Comment (domhanak) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/commands/CreateRepository.java]: Imho we should do something when `sslVerify` is true and repositories have the property already set to `false`.

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/JGitCloneTest.java [hunk 10]
```diff
@@ -246,6 +261,8 @@ public void cloneMirrorRepoConfigTest() throws IOException {
         assertEquals(Clone.REFS_MIRRORED, config.getString("remote", "origin", "fetch"));
         assertNull(config.getString("remote", "origin", "mirror"));
         assertEquals(gitSource.getAbsolutePath(), config.getString("remote", "origin", "url"));
+        assertEquals(sslVerify, config.getBoolean("http", null, "sslVerify", !sslVerify));
```
Comment (ederign) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/test/java/org/uberfire/java/nio/fs/jgit/JGitCloneTest.java]: ssl verify is ! here in order go guarantee the value read.

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/commands/CreateRepository.java [hunk 11]
```diff
@@ -104,6 +139,17 @@ public CreateRepository(final File repoDir,
         }
     }
 
+    private void setupSSLVerify(Repository repo) throws java.io.IOException {
+        StoredConfig config = repo.getConfig();
+        if (sslVerify) {
+            config.setBoolean("http", null, "sslVerify", true);
+            config.save();
+        } else {
+            config.setBoolean("http", null, "sslVerify", false);
```
Comment (ederign) [inline uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/commands/CreateRepository.java]: @domhanak

**Comment by ederign** [comment]
@domhanak thanks for the tests. PR updated.

image

The only scenario that is not covered is Scenario 4 - Not working as expected. All our boolean configurations default to false if the user passes an invalid parameter (Java API has the same behavior).

**Comment by ederign** [comment]
Jenkins execute full downstream build

File: uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/commands/CreateRepository.java (hunk 11 above)
Comment (jhrcek) [reply uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/commands/CreateRepository.java]: Nitpick The same could have been accomplished without if-else:
```java
config.setBoolean("http", null, "sslVerify", sslVerify);
config.save();
```

But don't let that block merging this PR.

**Comment by ederign** [comment]
jenkins retest this

**Pull request merged**